graph/telemetry/istio/appender/dead_node.go

```diff
@@ -32,11 +33,16 @@ func (a DeadNodeAppender) AppendGraph(trafficMap graph.TrafficMap, globalInfo *g
 		namespaceInfo.Vendor[workloadListKey] = &workloadList
 	}
 
-	a.applyDeadNodes(trafficMap, globalInfo, namespaceInfo)
+	// Apply dead node removal iteratively until no dead nodes are found.  Removal of dead nodes may
+	// alter the graph such that new nodes qualify for dead-ness by being orphaned, lack required
+	// outgoing edges, etc.. so we repeat as needed.
+	applyDeadNodes := true
+	for applyDeadNodes {
+		applyDeadNodes = a.applyDeadNodes(trafficMap, globalInfo, namespaceInfo) > 0
```

Comment:
Scary - if this a.applyDeadNodes code has a problem or some graph has some weirdness, this will lock the server in an infinite loop. For example, if applyDeadNodes fails to remove a dead node, over and over we go.

Reply:
well there are only so many nodes :)  If we remove all of them then we're done.   I'm not sure what you are suggesting, this is basically a while-loops, lots of loops, recursive algorithms, whatever, are hosed if the invariant is never satisfied.  I'm not adding some additional maxTries if that's what you are intoning.

Reply:
Not that it was necessary but to ensure I never have to listen to you whine about it, I added a check.